extensions/amp-byside-content/0.1/amp-byside-content.js

```diff
@@ -0,0 +1,346 @@
+      'label': this.label_,
+      'webcare_id': this.webcareId_,
+      'bwch': this.channel_ || '',
+      'lang': this.lang_ || '',
+      'fid': this.fid_ || '',
+      'bwit': (this.fid_ ? 'I' : 'A'),
+      'tuid': 'CLIENT_ID(byside_webcare_tuid)',
+      'suid': '',
+      'puid': 'PAGE_VIEW_IDpTIMESTAMP',
+      'referrer': 'DOCUMENT_REFERRER',
+      'page': 'SOURCE_URL',
+      'amppage': 'AMPDOC_URL',
+      'bwpt': 'TITLE',
+      'bres': 'VIEWPORT_WIDTHxVIEWPORT_HEIGHT',
+      'res': 'SCREEN_WIDTHxSCREEN_HEIGHT',
+      'v': 'v20171116a',
+      'ampv': 'AMP_VERSION',
+      'viewer': 'VIEWER',
+      'ua': 'USER_AGENT',
+      'r': 'RANDOM',
+      '_resize': '1',
+    });
+
+    return Services.urlReplacementsForDoc(this.element)
+        .expandAsync(addParamsToUrl(src, params));
+  }
+
+  /**
+   * @return {string} Returns a string to identify this tag. May not be unique
+   * if the element label is not unique.
+   * @private
+   */
+  getName_() {
+    let suffix = this.webcareId_ || '';
+    suffix += suffix && this.label_ ? ': ' : '';
+    suffix += this.label_ || '';
+
+    return 'amp-byside-content: ' + (suffix || '<unknown tag>');
+  }
+
+  /**
+   * @returns {!Element} Returns the overflow element
+   * @private
+   */
+  getOverflowElement_() {
+    const createElement = utils.getElementCreator(this.element.ownerDocument);
+
+    const overflow = createElement('div', 'bs-overflow',
+        createElement('div', 'bs-overflow-content',
+            createElement('i', 'bs-arrow-down')
+        ));
+    overflow.setAttribute('overflow', '');
+
+    return overflow;
+  }
+
+  /** @return {!Element} @private */
+  createBySideLoader_() {
+    const createElement = utils.getElementCreator(this.element.ownerDocument);
+
+    const loadingPlaceholder =
+      createElement('div', 'bs-loading-container',
+          createElement('div', 'bs-loading-animation')
+      );
+
+    return loadingPlaceholder;
+  }
+
+  /**
+   * Updates the element's dimensions to accommodate the iframe's
+   *    requested dimensions.
+   * @param {Object} data
+   * @private
+   */
+  updateSize_(data) {
+    // Calculate new width and height of the container to include the padding.
+    // If padding is negative, just use the requested width and height directly.
+    if (!data) {
+      return;
+    }
+
+    let newHeight, newWidth;
+    const height = parseInt(data['height'], 10);
+    if (!isNaN(height)) {
+      newHeight = Math.max(
+          height + (this.element./*OK*/offsetHeight
+              - this.iframe_./*OK*/offsetHeight),
+          height);
+    }
+    const width = parseInt(data['width'], 10);
+    if (!isNaN(width)) {
+      newWidth = Math.max(
+          width + (this.element./*OK*/offsetWidth
+              - this.iframe_./*OK*/offsetWidth),
+          width);
+    }
+
+    if (newHeight !== undefined || newWidth !== undefined) {
+      // Force change size as requested
+      this.element.getResources().attemptChangeSize(
```

Comment:
Trying to change `width` does not really work for most AMP layouts, specially responsive ones which are the most used. Please just use `this.attemptChangeHeight()`, content of your component should wrap properly to support any `width`, leaving only `height` as variable.